### PR TITLE
feat: delete link

### LINKED ISSUE #75: REST API
- [ ] Upload documents (and delete)
- [ ] Create link (and delete)
- [ ] Send link
- [ ] Retrieve views

Suggestions for additional functionality that an API should perform are welcomed

### PR DESCRIPTION
**Pull request by Aashish-Upadhyay-101**
fix #75

### CODE REVIEW

File: pages/api/links/[id]/index.ts [hunk 1]
```diff
@@ -128,6 +126,38 @@ export default async function handle(
     }
 
     return res.status(200).json(updatedLink);
+  } else if (req.method == "DELETE") {
```
Comment (mfts) [inline pages/api/links/[id]/index.ts:130]:
```suggestion
  } else if (req.method == "DELETE") {
  // DELETE /api/links/:id
```

File: pages/api/links/[id]/index.ts [hunk 2]
```diff
@@ -128,6 +126,38 @@ export default async function handle(
...
+    }
+
+    const { id } = req.query as { id: string };
+
+    try{
+      const linkToBeDeleted = await prisma.link.findUnique({
+        where: {
+          id: id,
+        }
+      });
```
Comment (mfts) [inline pages/api/links/[id]/index.ts]: I think we should add a way to determine that the user requesting to delete the link is actually authorized to delete the link. 

I like that you put this method in here. 
This allows us to compare the ownerId with the session userId, if you include the `document: { ownerId: true }`  on this function.

**Review by mfts — CHANGES_REQUESTED** [review]

**Comment by Aashish-Upadhyay-101** [comment]
@mfts changes done !

**Review by mfts — APPROVED** [review]
nice work @Aashish-Upadhyay-101 !

**Pull request merged**